src/measure_theory/interval_integral.lean

```diff
@@ -370,39 +370,110 @@ lemma integral_smul_measure (c : ℝ≥0∞) :
   ∫ x in a..b, f x ∂(c • μ) = c.to_real • ∫ x in a..b, f x ∂μ :=
 by simp only [interval_integral, measure.restrict_smul, integral_smul_measure, smul_sub]
 
-lemma integral_comp_add_right {a b : ℝ} (c : ℝ) {f : ℝ → E} (hfm : ae_measurable f) :
-  ∫ x in a..b, f (x + c) = ∫ x in a+c..b+c, f x :=
-have A : ae_measurable f (measure.map (λ x, x + c) volume), by rwa [real.map_volume_add_right],
-calc ∫ x in a..b, f (x + c) = ∫ x in a+c..b+c, f x ∂(measure.map (λ x, x + c) volume) :
-  by simp only [interval_integral, set_integral_map measurable_set_Ioc A (measurable_add_const _),
-    preimage_add_const_Ioc, add_sub_cancel]
-... = ∫ x in a+c..b+c, f x : by rw [real.map_volume_add_right]
+end basic
+
+section comp
 
-lemma integral_comp_mul_right {a b c : ℝ} {f : ℝ → E} (hc : 0 < c) (hfm : ae_measurable f) :
+variables {a b c : ℝ} (f : ℝ → E)
+
+lemma integral_comp_mul_right_of_pos (hc : 0 < c) :
   ∫ x in a..b, f (x * c) = c⁻¹ • ∫ x in a*c..b*c, f x :=
 begin
-  have A : ae_measurable f (measure.map (λ (x : ℝ), x*c) volume),
-    by { rw real.map_volume_mul_right (ne_of_gt hc), exact hfm.smul_measure _ },
+  have A : closed_embedding (λ (x : ℝ), x * c) :=
+    (homeomorph.mul_right' c (ne_of_gt hc)).closed_embedding,
   conv_rhs { rw [← real.smul_map_volume_mul_right (ne_of_gt hc)] },
   rw [integral_smul_measure],
-  simp only [interval_integral, set_integral_map measurable_set_Ioc A (measurable_mul_const _),
-    hc, preimage_mul_const_Ioc, mul_div_cancel _ (ne_of_gt hc), abs_of_pos,
+  simp only [interval_integral, hc, preimage_mul_const_Ioc, mul_div_cancel _ (ne_of_gt hc),
+    abs_of_pos, set_integral_map_of_closed_embedding measurable_set_Ioc A,
     ennreal.to_real_of_real (le_of_lt hc), inv_smul_smul' (ne_of_gt hc)],
 end
 
-lemma integral_comp_mul_left {a b c : ℝ} {f : ℝ → E} (hc : 0 < c) (hfm : ae_measurable f) :
+lemma integral_comp_neg : ∫ x in a..b, f (-x) = ∫ x in -b..-a, f x :=
+begin
+  have A : closed_embedding (λ (x : ℝ), - x) := (homeomorph.neg ℝ).closed_embedding,
+  conv_rhs { rw ← real.map_volume_neg },
+  simp only [interval_integral, set_integral_map_of_closed_embedding measurable_set_Ioc A,
+    neg_preimage, preimage_neg_Ioc, neg_neg, restrict_congr_set Ico_ae_eq_Ioc],
+end
+
+lemma integral_comp_mul_right_of_neg (hc : c < 0) :
+  ∫ x in a..b, f (x * c) = c⁻¹ • ∫ x in a*c..b*c, f x :=
+begin
+  let g := λ x, f (-x),
+  have h : (λ x, f (x * c)) = λ x, g (x * -c) := by simp_rw [g, neg_mul_eq_mul_neg, neg_neg],
+  rw [h, integral_comp_mul_right_of_pos g (neg_pos.mpr hc), integral_comp_neg f, integral_symm],
+  simp only [neg_mul_eq_mul_neg, neg_neg, inv_neg, neg_smul, ← smul_neg],
+end
+
+lemma integral_comp_mul_right (hc : c ≠ 0) :
+  ∫ x in a..b, f (x * c) = c⁻¹ • ∫ x in a*c..b*c, f x :=
```

Comment:
There is a variant of this lemma, 
```lean
  c • ∫ x in a..b, f (x * c) = ∫ x in a*c..b*c, f x
```
which works when `c = 0`, too.  Perhaps for a future PR?